packages/components/app/styles/components/side-nav/main.scss

```diff
@@ -66,56 +72,98 @@
 
 .hds-side-nav__menu-toggle-button {
   position: absolute;
+  top: -2px;
+  left: -1px;
   z-index: 1;
+  width: 24px;
+  height: 36px;
+  padding: 9px 0;
   color: var(--token-color-foreground-high-contrast);
   background: none;
-  border: 1px solid transparent;
+  background-color: var(--token-side-nav-color-surface-primary);
+  border: 1px solid var(--token-color-palette-neutral-500);
+  border-left-color: transparent;
+  border-top-right-radius: 5px;
+  border-bottom-right-radius: 5px;
+  transform:
+    translateX(var(--hds-app-sidenav-width-expanded))
+    translateY(24px);
   cursor: pointer;
-  transition-timing-function: var(--hds-app-sidenav-animation-easing);
-  transition-duration: var(--hds-app-sidenav-animation-duration);
-  transition-property: width, height, padding, border-radius, transform;
+  transition:
+    transform var(--hds-app-sidenav-animation-duration)
+    var(--hds-app-sidenav-animation-easing),
+    width var(--hds-app-sidenav-animation-duration)
+    var(--hds-app-sidenav-animation-easing);
+
+  &::before {
+    position: absolute;
+    top: -10px;
+    left: -1px;
+    z-index: -1;
+    width: 10px;
+    height: 10px;
+    color: var(--token-color-foreground-high-contrast);
+    border-bottom: 1px solid var(--token-color-palette-neutral-500);
+    border-left: 1px solid var(--token-color-palette-neutral-500);
+    border-bottom-left-radius: 5px;
+    box-shadow: 0 5px 0 var(--token-side-nav-color-surface-primary);
+    content: "";
+  }
+
+  &::after {
+    position: absolute;
+    bottom: -10px;
+    left: -1px;
+    width: 10px;
+    height: 10px;
+    color: var(--token-color-foreground-high-contrast);
+    border-top: 1px solid var(--token-color-palette-neutral-500);
+    border-left: 1px solid var(--token-color-palette-neutral-500);
+    border-top-left-radius: 5px;
+    box-shadow: 0 -5px 0 var(--token-side-nav-color-surface-primary);
+    content: "";
+  }
 
   &:hover {
```

Comment:
<img width="1220" alt="Screenshot 2023-09-15 at 5 05 22 PM" src="https://github.com/hashicorp/design-system/assets/2200899/8d5559c0-2512-4c3d-9cba-dbbcbfa452ad">

Small note on this, the intention with the icon is to "follow" the right edge of the button; the button expands in width on hover, but the icon also maintains the same distance for the right edge of the button. I'm trying to be a bit agnostic in how I refer to the animation because I'm not 100% sure how you have all of this built 😅 , could be a transform, or could be an alignment/flex/whatever property that aligns the icon to the end of the button container.

TLDR, the intention here is for the button to always maintain a 4px distance from the right edge of the button.

Reply:
Ah, good to know! I updated the code for it to behave as described above

Reply:
Looks great! Thanks for digging into that